tests/build/build_templates.go

```diff
@@ -35,6 +35,49 @@ var (
 
 const pipelineCompletionRetries = 2
 
+// OnboardComponent onboards a component from a test repository URL and returns the component's name
+func OnboardComponent(ctrl *has.HasController, gitUrl, revision, applicationName, componentName, namespace string) string {
```

Comment:
since we are creating only `cdq` which does not complete the onboarding till we create a component.. does this function name sounds a bit confusing.. will `PrepareForOnboardingComponent` sound better?